labshare-client/src/backend/endpoints/registration.ts

```diff
@@ -36,21 +44,42 @@ export async function registration(req: express.Request, res: express.Response,
 
     if (body.role === UserRoles.VOLUNTEER) {
         delete body.address.street
+        body.verified = { 
+            manually: true
+        }
     }
 
-    let user = await getUserForMail(body.contact.email)
-    if (user) {
+    if (OPT.DISABLE_VERIFICATION) {
+        body.verified = {
+            mail: true,
+            manually: true
+        }
+    }
+
+    let euser = await getUserForMail(body.contact.email)
+    if (euser) {
         return utils.errorResponse(res, HttpStatus.BAD_REQUEST, "existing_user")
     }
 
     let model = getModelForRole(role)
     if (!model) return utils.badRequest(res)
-    let doc = new model(body)
+    let user = new model(body)
+    
+    let token = uuid();
+    let token_doc = new ActivationToken({ token: token, objectId: user._id });
 
-    doc.save().then((doc) => {
-        if (!doc) {
-            return utils.internalError(res)
-        }
+    user.save().then((doc) => {
```

Comment:
This promise block can be `awaited` and put in a `try/catch` without having to change anything major